worldedit-core/src/main/java/com/sk89q/worldedit/util/collection/VectorPositionList.java

```diff
@@ -0,0 +1,52 @@
+package com.sk89q.worldedit.util.collection;
+
+import com.google.common.collect.AbstractIterator;
+import com.sk89q.worldedit.math.BlockVector3;
+import it.unimi.dsi.fastutil.objects.ObjectArrayList;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+
+class VectorPositionList implements PositionList {
+
+    private final List<BlockVector3> delegate = new ObjectArrayList<>();
```

Comment:
Maybe an IntList and just keep triples of ints? That should be only 50% more memory than LongList, instead of over twice as much.

Reply:
Done.